compound.py

```diff
@@ -137,72 +138,141 @@ def update_bot_order_volumes(
 ):
     """Update bot with new order volumes."""
 
+    bot_id = thebot["id"]
     bot_name = thebot["name"]
     base_order_volume = float(thebot["base_order_volume"])
     safety_order_volume = float(thebot["safety_order_volume"])
+    #### I know these are also down below, but to make calculation easier, I'm recalculating these
+    bo_profit = new_base_order_volume - base_order_volume
+    so_profit = new_safety_order_volume - safety_order_volume
+    #### Check from the database what the last stored values are
+    #### db_lastpassupdate = was the last update to BO executed or stored?
+    #### db_lastcalcbo = Last Calculated BO from Database
+    #### db_lastcalcso = Last Caclulated SO from Database
+    db_lastpassupdate = cursor.execute(
+        f"SELECT lastpassupdate FROM bots WHERE botid = {bot_id}"
+    ).fetchone()[0]
+    logger.debug(
+        f"lastpassupdate value in db is {db_lastpassupdate}"
+    )
+    db_lastcalcbo = cursor.execute(
+        f"SELECT lastcalcbo FROM bots WHERE botid = {bot_id}"
+    ).fetchone()[0]
+    logger.debug(
+        f"lastcalcbo value in db is {db_lastcalcbo}"
+    )
+    db_lastcalcso = cursor.execute(
+        f"SELECT lastcalcso FROM bots WHERE botid = {bot_id}"
+    ).fetchone()[0]
+    logger.debug(
+        f"lastcalcso value in db is {db_lastcalcso}"
+    )
 
+    #### Check if the BO/SO was updated on the last pass
+    #### If not, add the value from the DB to the new BO
+    if db_lastpassupdate == 'No':
+        new_base_order_volume = (db_lastcalcbo + bo_profit)
+        new_safety_order_volume = (db_lastcalcso + so_profit)
+        logger.debug(
+            f"bo_profit is ({bo_profit})\n"
+            f"so_profit is ({so_profit})\n"
+            f"new_base_order_volume ({new_base_order_volume}) is db_lastcalcbo ({db_lastcalcbo}) + bo_profit ({bo_profit})\n"
+            f"new_safety_order_volume ({new_safety_order_volume}) is db_lastcalcso ({db_lastcalcso}) + so_profit ({so_profit})"
```

Comment:
Dan was je zo blij met de leesbare values, en print je het hier op de oude manier :D Je kunt hier dus de -e06 values krijgen, je kunt even bij de andere logging kijken hoe je dat kunt voorkomen. Het is wel enkel debug log dus minder wat een gebruiker ziet, maar toch.

Reply:
Ja, dat was me ook opgevallen 😂 
Aangepast